### ISSUE TITLE
Error when enabling Future Parser (Puppet 3.7.3 Server and Client)

### ISSUE THREAD

**Issue by sharkannon** [pall-valmundsson/puppet-createrepo#12]
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Found 2 errors. Giving up in /etc/puppet/environments/production/modules/createrepo/manifests/init.pp on node caondvyum01.caondv.local
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run

After some troubleshooting, it appears that the futureparser doesn't like the following statement:
```ruby
if type($timeout) != 'integer' { fail('$timeout is not an integer') }
```

**Comment by sharkannon** [comment]
```ruby
if $changelog_limit =~ /^\d+$/
```
Causes errors once I fixed that specific issue.  The regular expression can only check a STRING and not an integer.  Need to convert $changelog_limit to string if you want to use this exact condition.

**Comment by pall-valmundsson** [comment]
Thanks for the report. I just added future parser testing for 3.7 to the Travis config. I'll look into this.

**Comment by pall-valmundsson** [comment]
Ok. I got testing working. But somehow I'm unable to reproduce the issue https://travis-ci.org/pall-valmundsson/puppet-createrepo/jobs/48030948

**Comment by sharkannon** [comment]
Reproduce the original issue? Puppet 3.7.3 with Future Parser

**Comment by sharkannon** [comment]
Actually, just looked at your output.. I see you're using 3.7.0, but I doubt that should be an issue.  Not sure WHY you can't reproduce :( Hrm

**Comment by pall-valmundsson** [comment]
Yeah. I welcomed an actual issue with future parser so that I could configure the tests and verifiably fail it, and then fix it. I'll try a manual run with 3.7.3 and future parser.

**Comment by sharkannon** [comment]
FYI, Ruby version is 1.8,7 (Centos 6.5).. wonder if that'll make a diff

**Comment by pall-valmundsson** [comment]
Ok. And that introduces another issue as travis (at least my config) seems to have issues installing the test dependencies with 1.8.7. OTOH my production setup is running with 1.8.7 so I'll also need to look into that.

**Comment by sharkannon** [comment]
I understand man, this is the first issue I've had with FP enabled.  only reason I even use it is because of 1 requirement in 1 module I have that I HAVE to use a FOR loop in (Executing SQL scripts in mysql)

**Comment by sharkannon** [comment]
Ahh, I haven't touched Travis CI, implemnting my own tests in Jenkins with puppet parser verify and puppet-lint atm, haven't written a lot of test cases yet due to the fact I'm still learning :)

**Comment by pall-valmundsson** [comment]
Can you paste a manifest that has the issue? I tried `puppet parser validate validate.pp --parser=future --debug` where validate.pp is the basic example from the README without errors on RHEL6 Ruby 1.8.7 and puppet 3.7.3...

**Comment by sharkannon** [comment]
```ruby
# Class: repos
#
# This module manages repos
#
# Parameters: none
#
# Actions:
#
# Requires: see Modulefile
#
# Sample Usage:
#
class repos($repouser = 'repouser', $repogroup='repogroup') {
  File {
    owner => $repouser,
    group => $repogroup,
  }

  Createrepo {
    repo_owner => $repouser,
    repo_group => $repogroup,
  }

  file { ['/var/yumrepos', '/var/cache/yumrepos']: ensure => 'directory', }

  file { ['/var/yumrepos/dev', '/var/cache/yumrepos/dev']:
    ensure  => 'directory',
    require => File['/var/yumrepos', '/var/cache/yumrepos'],
  }

  file { ['/var/yumrepos/rel', '/var/cache/yumrepos/rel']:
    ensure  => 'directory',
    require => File['/var/yumrepos', '/var/cache/yumrepos'],
  }

  file { ['/var/yumrepos/dev/5', '/var/cache/yumrepos/dev/5']:
    ensure  => 'directory',
    require => File['/var/yumrepos/dev', '/var/cache/yumrepos/dev'],
  }

  file { ['/var/yumrepos/dev/6', '/var/cache/yumrepos/dev/6']:
    ensure  => 'directory',
    require => File['/var/yumrepos/dev', '/var/cache/yumrepos/dev'],
  }

  file { ['/var/yumrepos/rel/5', '/var/cache/yumrepos/rel/5']:
    ensure  => 'directory',
    require => File['/var/yumrepos/rel', '/var/cache/yumrepos/rel'],
  }

  file { ['/var/yumrepos/rel/6', '/var/cache/yumrepos/rel/6']:
    ensure  => 'directory',
    require => File['/var/yumrepos/rel', '/var/cache/yumrepos/rel'],
  }

  createrepo { 'yumrepo_dev_6_x86_64':
    repository_dir => '/var/yumrepos/dev/6/x86_64',
    repo_cache_dir => '/var/cache/yumrepos/dev/6/x86_64',
    enable_cron    => false,
    require        => File['/var/yumrepos/dev/6', '/var/cache/yumrepos/dev/6']
  }

  createrepo { 'yumrepo_dev_6_i386':
    repository_dir => '/var/yumrepos/dev/6/i386',
    repo_cache_dir => '/var/cache/yumrepos/dev/6/i386',
    enable_cron    => false,
    require        => File['/var/yumrepos/dev/6', '/var/cache/yumrepos/dev/6'],
  }

  createrepo { 'yumrepo_dev_5_x86_64':
    repository_dir => '/var/yumrepos/dev/5/x86_64',
    repo_cache_dir => '/var/cache/yumrepos/dev/5/x86_64',
    checksum_type  => 'sha1',
    enable_cron    => false,
    require        => File['/var/yumrepos/dev/5', '/var/cache/yumrepos/dev/5'],
  }

  createrepo { 'yumrepo_dev_5_i386':
    repository_dir => '/var/yumrepos/dev/5/i386',
    repo_cache_dir => '/var/cache/yumrepos/dev/5/i386',
    checksum_type  => 'sha1',
    enable_cron    => false,
    require        => File['/var/yumrepos/dev/5', '/var/cache/yumrepos/dev/5'],
  }

  createrepo { 'yumrepo_rel_6_x86_64':
    repository_dir => '/var/yumrepos/rel/6/x86_64',
    repo_cache_dir => '/var/cache/yumrepos/rel/6/x86_64',
    require        => File['/var/yumrepos/rel/6', '/var/cache/yumrepos/rel/6'],
  }

  createrepo { 'yumrepo_rel_6_i386':
    repository_dir => '/var/yumrepos/rel/6/i386',
    repo_cache_dir => '/var/cache/yumrepos/rel/6/i386',
    require        => File['/var/yumrepos/rel/6', '/var/cache/yumrepos/rel/6'],
  }

  createrepo { 'yumrepo_rel_5_x86_64':
    repository_dir => '/var/yumrepos/rel/5/x86_64',
    repo_cache_dir => '/var/cache/yumrepos/rel/5/x86_64',
    checksum_type  => 'sha1',
    require        => File['/var/yumrepos/rel/5', '/var/cache/yumrepos/rel/5'],
  }

  createrepo { 'yumrepo_rel_5_i386':
    repository_dir => '/var/yumrepos/rel/5/i386',
    repo_cache_dir => '/var/cache/yumrepos/rel/5/i386',
    checksum_type  => 'sha1',
    require        => File['/var/yumrepos/rel/5', '/var/cache/yumrepos/rel/5'],
  }

  class { 'nginx':
  }

  nginx::resource::vhost { "${::ipaddress}:8080":
    require     => File['/var/yumrepos/dev', '/var/cache/yumrepos/dev'],
    www_root    => '/var/yumrepos/dev',
    listen_port => 8080
  }

  nginx::resource::vhost { "${::ipaddress}:8081":
    require     => File['/var/yumrepos/rel', '/var/cache/yumrepos/rel'],
    www_root    => '/var/yumrepos/rel',
    listen_port => 8081
  }
}

**Comment by pall-valmundsson** [comment]
I managed to reproduce the errors with test and verify that your changes fixed those issues. Thanks a lot for you help.

**Comment by sharkannon** [comment]
Good to hear, was worried it was more unique to me, which would have meant I have a strange issue
